soql-server-pg/src/main/scala/com/socrata/pg/query/DataSqlizerQuerier.scala

```diff
@@ -116,13 +117,37 @@ trait DataSqlizerQuerier[CT, CV] extends AbstractRepBasedDataSqlizer[CT, CV] {
   def setTimeout(timeoutMs: String) = s"SET LOCAL statement_timeout TO $timeoutMs"
   def resetTimeout = "SET LOCAL statement_timeout TO DEFAULT"
 
-  private def executeSql(conn: Connection, pSql: ParametricSql, timeout: Option[Duration], fetchSize: Integer, debug: Boolean): ResultSet = {
+  def setContext(conn: Connection, context: Map[String, String]) {
+    if(context.nonEmpty) {
+      // Although in my experiments it looks like an arbitrary string
+      // works fine for the variable name, I'm md5ing it because I'm
+      // not willing to expose our databases to potential hostile
+      // activity in a thing that's not generally considered part of
+      // the query path.
+      //
+      // Because these things are mainly/exclusively set by core, I'm
+      // not going to care overmuch about the values' trustworthiness.
+      // This is mostly to prevent hypothetical attacks via the
+      // get_context function.
+      using(conn.prepareStatement("select set_config('user_variables.' || md5(?), ?, true)")) { stmt =>
```

Comment:
not sure if it is worthwhile to group that into batch of 5 so it can set multiples in one trip.

Reply:
It would not be difficult to do, so therefore I shall do it!  (Though I'll be surprised if we ever send more than one or two in practice)